components/compositing/compositor_thread.rs

```diff
@@ -76,16 +76,16 @@ pub enum Msg {
     ScrollFragmentPoint(PipelineId, ScrollRootId, Point2D<f32>, bool),
     /// Alerts the compositor that the current page has changed its title.
     ChangePageTitle(PipelineId, Option<String>),
-    /// Alerts the compositor that the current page has changed its URL.
-    ChangePageUrl(PipelineId, ServoUrl),
     /// Alerts the compositor that the given pipeline has changed whether it is running animations.
     ChangeRunningAnimationsState(PipelineId, AnimationState),
     /// Replaces the current frame tree, typically called during main frame navigation.
     SetFrameTree(SendableFrameTree, IpcSender<()>),
-    /// The load of a page has begun: (can go back, can go forward).
-    LoadStart(bool, bool),
-    /// The load of a page has completed: (can go back, can go forward, is root frame).
-    LoadComplete(bool, bool, bool),
+    /// The load of a page has begun
+    LoadStart,
+    /// The load of a page has completed
+    LoadComplete,
+    /// The history of a top level browser has changed.
+    HistoryChanged(HistoryEntries),
```

Comment:
Is this sending the joint session history?

Sending the whole history every time seems a bit wasteful, most of it won't have changed.